src/Core/Resolvers/Factories/MutationEngineFactory.cs

```diff
@@ -18,7 +19,15 @@ namespace Azure.DataApiBuilder.Core.Resolvers.Factories
     /// </summary>
     public class MutationEngineFactory : IMutationEngineFactory
     {
-        private readonly Dictionary<DatabaseType, IMutationEngine> _mutationEngines;
+        private Dictionary<DatabaseType, IMutationEngine> _mutationEngines;
+        private readonly RuntimeConfigProvider _runtimeConfigProvider;
```

Comment:
before calling the ConfigureMutationEngine function we create a new dictionary for _mutationEngines. That creation of a new dictionary isn't done inside of the configure function because it needs to be set in the constructor since it is non null.